helpers/setup-spark-k8s.py

```diff
@@ -0,0 +1,135 @@
+    token_expiry_duration='24h' # NOTE: The server may return a token with a longer or shorter lifetime.
+    os.system('{} create token {} --namespace {} --duration {} --kubeconfig={} --cluster={}'.format(kubectl_cmd, serviceaccountname, namespace, token_expiry_duration, kubeconfig, cluster_name))
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--kubeconfig", default='no-kubeconfig-provided', help='Kubernetes configuration file')
+    parser.add_argument("--cluster", default='no-cluster-provided', help='Cluster name to use within the provided kubernetes configuration file')
+    subparsers = parser.add_subparsers(dest='action')
+    subparsers.required = True
+
+    #  subparser for service-account
+    parser_account = subparsers.add_parser('service-account')
+    parser_account.add_argument('username', help='Service account username to be created in kubernetes.')
+    parser_account.add_argument('--namespace', default='default', help='Namespace for the service account to be created in kubernetes. Default is default namespace')
+
+    #  subparser for CA certificate
+    parser_certificate = subparsers.add_parser('get-ca-cert')
+
+    #  subparser for OAuth token
+    parser_token = subparsers.add_parser('get-token')
+    parser_token.add_argument('serviceaccount', help='Service account name for which the Oauth token is to be fetched.')
+    parser_token.add_argument('--namespace', default='default', help='Namespace for the service account for which the Oauth token is to be fetched. Default is default namespace')
+
+    args = parser.parse_args()
+
+    if args.kubeconfig != 'no-kubeconfig-provided':
+        kubeconfig = args.kubeconfig
+    else:
+        kubeconfig = '/home/{}/.kube/config'.format(os.getlogin())
+
+    if args.cluster != 'no-cluster-provided':
+        cluster_name = args.cluster
+    else:
+        cluster_name = try_extract_default_cluster_from_kubeconfig(kubeconfig)
+
+    if args.action == 'service-account':
+        username=args.username
+        namespace = args.namespace
+        # print ('kubeconfig={} cluster={} username={}, namespace={}'.format(kubeconfig, cluster_name, username, namespace))
+        setup_user(username, namespace, kubeconfig, cluster_name)
+    elif args.action == 'get-ca-cert':
+        # print ('kubeconfig={} cluster={}'.format(kubeconfig, cluster_name))
+        try:
+            extract_ca_crt_from_kube_config(kubeconfig, cluster_name)
+        except IOError as e:
+            print_help_for_missing_or_inaccessible_kubeconfig_file()
+            sys.exit(-400)
+    elif args.action == 'get-token':
+        serviceaccountname=args.serviceaccount
+        namespace = args.namespace
+        # print ('kubeconfig={} cluster={} serviceaccount={}, namespace={}'.format(kubeconfig, cluster_name, serviceaccountname, namespace))
+        generate_token(serviceaccountname, namespace, kubeconfig, cluster_name)
```

Comment:
I believe we should auth via kubeconfig, so content for `spark.kubernetes.authenticate.submission.*` variables is not needed at all.